src/web/handlers.rs

```diff
@@ -355,7 +355,16 @@ pub async fn post_collection_batch(
     match result {
         Ok(_) => success.extend(bso_ids),
         Err(e) if e.is_conflict() => return Err(e.into()),
-        Err(_) => failed.extend(bso_ids.into_iter().map(|id| (id, "db error".to_owned()))),
+        Err(apperr) => {
+            if let ApiErrorKind::Db(dberr) = apperr.kind() {
+                // If we're over quota, return immediately with a 403 to let the client know.
+                // Otherwise the client will simply keep retrying records.
+                if let DbErrorKind::Quota = dberr.kind() {
+                    return Ok(apperr.into());
```

Comment:
Shouldn't this return Err instead of Ok? Though I'm not sure it makes a difference.

We might wanna make sure this is tested which can be added later during #120